tests/core/test_structured_data_signing.py

```diff
@@ -237,9 +234,8 @@ def test_invalid_structured_data_invalid_abi_type():
         "tests/fixtures/invalid_message_invalid_abi_type.json"
     ).read()
     invalid_structured_data = json.loads(invalid_structured_data_string)
-    with pytest.raises(ABITypeError) as e:
+    with pytest.raises(TypeError, match="Received Invalid type `uint25689` in the struct `Person`"):
```

Comment:
This is a small breaking change, but decided to go for it since this release will be breaking with the dependency updates anyways. `eth-abi` had a bug where `ABITypeError` was raised on a `has_encoder` check instead of returning `False`. That got fixed downstream in `eth-abi`, which caused this test to fail. Raising this `TypeError` here is consistent with other validation checks for signing structured data.

Reply:
Yeah makes sense... I'm good with it 👍